Tests/SwiftStdlibTests/DictionaryExtensionsTests.swift

```diff
@@ -78,6 +78,15 @@ final class DictionaryExtensionsTests: XCTestCase {
         XCTAssertEqual(dict, ["testkey": "value"])
     }
 
+    func testSubscriptKeypath() {
+        var json = ["key": ["key1": ["key2": "value"]]]
+
+        XCTAssertEqual(json[path: []] as? String, nil)
```

Comment:
Also a test case for `["key", "key1"]` returning `["key2": "value"]` can be a good addition :))